src/views/bibliography_item.tsx

```diff
@@ -44,14 +50,56 @@ const createBibliography = async (items: BibliographyItem[]) => {
   return contents
 }
 
-interface BibliographyItemProps extends BaseNodeProps {
+interface BibliographyItemViewProps extends BaseNodeProps {
   setComment: (comment?: CommentAnnotation) => void
+  filterLibraryItems: (query: string) => Promise<BibliographyItem[]>
+  saveModel: <T extends Model>(model: T | Build<T> | Partial<T>) => Promise<T>
+  deleteModel: (id: string) => Promise<string>
+  setLibraryItem: (item: BibliographyItem) => void
+  removeLibraryItem: (id: string) => void
+  modelMap: Map<string, Model>
+  components: Record<string, React.ComponentType<any>> // eslint-disable-line @typescript-eslint/no-explicit-any
 }
 
-export class BibliographyItemView<PropsType extends BibliographyItemProps>
+export class BibliographyItemView<PropsType extends BibliographyItemViewProps>
   extends BaseNodeView<PropsType>
   implements ManuscriptNodeView
 {
+  protected popperContainer?: HTMLDivElement
+
+  public showPopper = (referenceID: string) => {
+    const {
+      filterLibraryItems,
+      saveModel,
+      deleteModel,
+      setLibraryItem,
+      removeLibraryItem,
+      renderReactComponent,
+      modelMap,
+      components: { ReferencesEditor },
+    } = this.props
+
+    if (!this.popperContainer) {
+      this.popperContainer = document.createElement('div')
+      this.popperContainer.className = 'references'
+    }
+
+    renderReactComponent(
+      <ReferencesEditor
+        filterLibraryItems={filterLibraryItems}
+        saveModel={saveModel}
+        deleteModel={deleteModel}
+        setLibraryItem={setLibraryItem}
+        removeLibraryItem={removeLibraryItem}
+        modelMap={modelMap}
+        referenceID={referenceID}
+      />,
+      this.popperContainer
+    )
+
+    this.props.popper.show(this.dom, this.popperContainer, 'right')
```

Comment:
should we destroy the popper if it's closed? as I still face issue with opening it for the second time

Reply:
look at https://www.loom.com/share/f069755eaefc4c78a6f660a60681d237

Reply:
Sounds yes, I should.

Reply:
I tried the destroy but did not work, setting popperContainer to undefined, fix the issue for me